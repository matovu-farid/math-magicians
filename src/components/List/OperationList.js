import propTypes from 'prop-types';
import { Component } from 'react';
import { OpperationTile } from '../Tiles/Tile';

class OperationList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.symbolArray = ['รท', '-', 'x', '+', '='];
  }

  render() {
    const { handleButtonClick, handleOpperate } = this.props;
    return (
      <ul className="opp-list">

        {this.symbolArray
          .map((symbol) => (
            <OpperationTile
              key={symbol}
              opp={symbol}
              handleButtonClick={handleButtonClick}
              handleOpperate={handleOpperate}
            />
          ))}
      </ul>
    );
  }
}

OperationList.propTypes = {
  handleButtonClick: propTypes.func.isRequired,
  handleOpperate: propTypes.func.isRequired,
};

export default OperationList;
