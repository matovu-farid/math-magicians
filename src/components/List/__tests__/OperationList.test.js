import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import OperationList from '../OperationList';

it('renders correctly', () => {
  const tree = renderer.create(<OperationList />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should have 5 buttons for operations', () => {
  render(<OperationList />);
  const listItems = screen.queryAllByRole('listitem');

  expect(listItems.length).toBe(5);
});
