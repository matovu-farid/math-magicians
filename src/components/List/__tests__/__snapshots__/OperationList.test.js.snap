// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="opp-list"
>
  <li>
    <button
      className="tile orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </li>
  <li>
    <button
      className="tile orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </li>
  <li>
    <button
      className="tile orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </li>
  <li>
    <button
      className="tile orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </li>
  <li>
    <button
      className="tile orange"
      type="button"
    >
      =
    </button>
  </li>
</ul>
`;
